snapcraft/extensions/basic_graphics.py

```diff
@@ -0,0 +1,134 @@
+    def is_experimental(base: Optional[str]) -> bool:
+        return False
+
+    @overrides
+    def get_root_snippet(self) -> Dict[str, Any]:
+        base = self.yaml_data["base"]
+        return {
+            "plugs": {
+                _GRAPHICS_NAME[base]: {
+                    "interface": "content",
+                    "target": "$SNAP/graphics",
+                    "default-provider": _MESA_SNAP[base],
+                },
+            },
+            "layout": {
+                "/usr/share/drirc.d": {
+                    "bind": "$SNAP/graphics/drirc.d",
+                },
+                "/usr/share/libdrm": {
+                    "bind": "$SNAP/graphics/libdrm",
+                },
+                "/usr/share/X11": {
+                    "bind": "$SNAP/graphics/X11",
+                },
+            },
+        }
+
+    @overrides
+    def get_app_snippet(self) -> Dict[str, Any]:
+        return {
+            "command-chain": [
+                "bin/graphics-launch",
+            ],
+            "plugs": ["opengl", "wayland", "x11"],
+        }
+
+    @overrides
+    def get_part_snippet(self, *, plugin_name: str) -> Dict[str, Any]:
+        return {
+            "build-packages": [
+                "libdrm-dev",
+                "libegl-dev",
+                "libgbm-dev",
+                "libgl-dev",
+                "libgles-dev",
+                "libglx-dev",
+                "libnvidia-egl-wayland-dev",
+                "libva-dev",
+                "libvdpau-dev",
+                "libvulkan-dev",
+                "libwayland-dev",
+                "libwayland-egl-backend-dev",
+                "libx11-xcb-dev",
+                "libxau-dev",
+                "libxcb-dri2-0-dev",
+                "libxcb-dri3-dev",
+                "libxcb-present-dev",
+                "libxcb-sync-dev",
+                "libxcb-xfixes0-dev",
+                "libxcb1-dev",
+                "libxdamage-dev",
+                "libxdmcp-dev",
+                "libxshmfence-dev",
+            ],
+        }
+
+    @overrides
+    def get_parts_snippet(self) -> Dict[str, Any]:
+        base = self.yaml_data["base"]
+        return {
+            f"{_EXT_NAME}/{_EXT_NAME}": {
+                "source": str(get_extensions_data_dir() / _EXT_NAME),
+                "plugin": "dump",
+                "override-build": f"sed -i 's/%GRAPHICS%/{_GRAPHICS_NAME[base]}/g' $CRAFT_PART_BUILD/bin/graphics-launch\n"
+                                  f"sed -i 's/%MESA%/{_MESA_SNAP[base]}/g' $CRAFT_PART_BUILD/bin/graphics-launch\n"
+                                  "craftctl default",
+                "override-prime": "craftctl default\n${CRAFT_PART_SRC}"
+                                  f"/bin/graphics-cleanup {_MESA_SNAP[base]}",
+                "prime": ["bin/graphics-launch"],
+                # This should be the last part to prime, so the clean-up can execute properly
+                "after": list(self.yaml_data["parts"].keys()),
```

Comment:
That may be the case, though as I understand the naming convention for extension-provided parts the `extname/` prefix is required.  I've no idea if applying an ordering prefix to the portion after the `/` would be applied in the same way

Reply:
FWIW I mean that there might be an extension processed _after_ yours, in which case the `after:` will miss any parts introduced there.

It may be needed to document that this extension should be last, if that actually matters.